client/packages.go

```diff
@@ -176,8 +183,9 @@ func (client *Client) Find(opts *FindOptions) ([]*Snap, *ResultInfo, error) {
 	case opts.Private:
 		q.Set("select", "private")
 	}
-	if opts.Section != "" {
-		q.Set("section", opts.Section)
+
+	if section != "" || (section == "" && query == "") {
```

Comment:
Maybe I misunderstand what you mean by "backwards-compatible way".

The overall semantics (being reflected there) is: an empty search should not fail and return the list of featured snaps.

In the store (I agree that this is an implementation detail) it is being translated as a query of the form: search?q=&section=

(this has not the same semantics as search?q= which is supposed to list all the snaps)

This reflects that (I dont mean that it has to be that way, I am far from satisfied in order to keep the logic as it is in snapd's client, the daemon, and the store backend ... a proper refactoring can be done after),

As a side note, the snap find command already has some logic along those lines:

https://github.com/snapcore/snapd/blob/master/cmd/snap/cmd_find.go#L108

This will need to be tweaked abit though, I am not super fan of the hardcoded string, but the semantics is somewhat similar to the one intended above: return the list of featured snaps.


This will need to be tweaked abit though, I am not super fan of the hardcoded string, but the semantics is somewhat similar to the one intended above: return the list of featured snaps.

Thinking about it, yes, we dont have to reflect that store specific request "encoding" (a bit overloaded too), ... OK, I'll update that for now, it is too convoluted like that